### ISSUE TITLE
Snippets dont work as intended

### ISSUE THREAD

**Issue by marwan38** [NvChad/NvChad#436]
**Describe the bug**
Snippets don't work when you start off with typing `p`, for example, or just press tab to move onto the next value. I'm not sure if latter is an NvChad thing but I thought i'd post it here together to, hopefully, get some insight.

**To Reproduce**
1. Create file: test.php
2. Find snippet called con to bring out the constructor snippet
3a. Try typing private or public
3b. Try pressing tab to skip the first snippet variable

**Expected behavior**
I expect that when I press `p` when the snippet is expanded, vim goes into insert mode and replaces `public` with `p`.
I also expect that If I press tab, it skips the first snippet value, `public` in this example, and moves onto the next value.

**Additional context**
This was replicated with a barebones config using the nvchad plugins for cmp and snippets and the plugin config for `cmp` from `NvChad`. The first issue was solved and the expected behavior was the result.
Discord conversation: https://discord.com/channels/869557815780470834/869561890672635985/884701509932158976

**Comment by G-Rowell** [comment]
Hmmm,

@marwan38 still having this issue?
I'm happy to investigate it this week?

**Comment by marwan38** [comment]
@G-Rowell I migrated to my own config for the time being. It's easy to replicate, if you want to try again.

**Comment by siduck76** [comment]
I feel like this an upstream luasnip issue ._.

**Comment by marwan38** [comment]
> I feel like this an upstream luasnip issue ._.

I don't think so, i've had it working just fine in every other config I've tried, including and bare minimum. I plan on coming back to NvChad soon, so we'll see then :)

**Comment by siduck76** [comment]
> > I feel like this an upstream luasnip issue ._.
> 
> I don't think so, i've had it working just fine in every other config I've tried, including and bare minimum. I plan on coming back to NvChad soon, so we'll see then :)

make sure you check this before https://nvchad.github.io/config
